library/src/com/google/maps/android/kml/KmlContainerParser.java

```diff
@@ -0,0 +1,204 @@
+
+    /* package */ void createContainer() throws XmlPullParserException, IOException {
+        KmlContainer kmlContainer = new KmlContainer();
+        assignContainerProperties(kmlContainer);
+        mContainer = kmlContainer;
+    }
+
+    /**
+     * Obtains relevant values from an XML start tag and assigns these values to variables
+     * within a KmlContainer class
+     *
+     * @param kmlContainer Container to store XML start tags and its corresponding value
+     */
+    /* package */ void assignContainerProperties(KmlContainer kmlContainer)
+            throws XmlPullParserException, IOException {
+        String startTag = mParser.getName();
+
+        if (mParser.getAttributeValue(null, "id") != null) {
+            kmlContainer.setContainerId(mParser.getAttributeValue(null, "id"));
+        }
+
+        mParser.next();
+        int eventType = mParser.getEventType();
+        while (!(eventType == END_TAG && mParser.getName().equals(startTag))) {
+            if (eventType == START_TAG) {
+                if (mParser.getName().matches(CONTAINER_REGEX)) {
+                    setNestedContainerObject(kmlContainer);
+                } else if (mParser.getName().matches(PROPERTY_REGEX)) {
+                    setContainerProperty(kmlContainer);
+                } else if (mParser.getName().equals(STYLE_MAP)) {
+                    setContainerStyleMap(kmlContainer);
+                } else if (mParser.getName().equals(STYLE)) {
+                    setContainerStyle(kmlContainer);
+                } else if (mParser.getName().equals(PLACEMARK)) {
+                    setContainerPlacemark(kmlContainer);
+                } else if (mParser.getName().equals(EXTENDED_DATA)) {
+                    setExtendedDataProperties(kmlContainer);
+                } else if (mParser.getName().equals(GROUND_OVERLAY)) {
+                    mFeatureParser.createGroundOverlay();
+                    KmlGroundOverlay kmlGroundOverlay = mFeatureParser.getGroundOverlay();
+                    kmlContainer.addGroundOverlay(kmlGroundOverlay);
+                }
+            }
+            eventType = mParser.nextToken();
+        }
+    }
+
+    /**
+     * Creates a new Container object (created if a Document or Folder start tag is read by the
+     * XmlPullParser) and assigns specific elements read from the XmlPullParser to the container.
+     *
+     * @param kmlFolder Stores new container object
+     */
+    /* package */ void setNestedContainerObject(KmlContainer kmlFolder)
+            throws XmlPullParserException, IOException {
+        KmlContainer container = new KmlContainer();
+        assignContainerProperties(container);
+        kmlFolder.addChildContainer(container);
+    }
+
+    /**
+     * Creates a new style map and assigns values from the input parser
+     * and stores it into the container.
+     *
+     * @param kmlContainer Stores new style map
+     */
+    /* package */ void setContainerStyleMap(KmlContainer kmlContainer)
```

Comment:
I find it a little weird that these methods take in KmlContainer as a parameter but use the mParser field.  It makes it look like these are independent when I suspect they are not.